jazz_deployments/components/utils.js

```diff
@@ -0,0 +1,239 @@
+// =========================================================================
+// Copyright © 2017 T-Mobile USA, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =========================================================================
+
+/**
+    Helper functions for Environment
+    @module: utils.js
+    @description: Defines functions like format the output as per Environment schema etc.
+    @author:
+    @version: 1.0
+**/
+
+var AWS = require("aws-sdk");
+
+// initialize document CLient for dynamodb
+var initDocClient = function() {
+    AWS.config.update({ region: global.config.ddb_region });
+    var docClient = new AWS.DynamoDB.DocumentClient();
+
+    return docClient;
+};
+
+var initDynamodb = function() {
+    AWS.config.update({ region: global.config.ddb_region });
+    var dynamodb = new AWS.DynamoDB();
+
+    return dynamodb;
+};
+
+var isEmpty = function(obj) {
+    for (var prop in obj) {
+        if (obj.hasOwnProperty(prop)) return false;
+    }
+    return true;
+};
+
+// convert object returned from the database, as per schema
+var formatData = function(data, format) {
+    if (data === undefined || data === null) {
+        return {};
+    }
+
+    var deployment_obj = {};
+
+    var parseValue = function(value) {
+        var type = Object.keys(value)[0];
+        var parsed_value = value[type];
+        if (type === "NULL") {
+            return null;
+        } else if (type === "N") {
+            return Number(value);
+        } else if (type === "NS") {
+            return parsed_value.map(Number);
+        } else if (type === "S") {
+            return parsed_value;
+        } else if (type === "SS") {
+            return parsed_value;
+        } else if (type === "L") {
+            var parsed_value_list = [];
+            try {
+                for (var i = 0; i < parsed_value.length; i++) {
```

Comment:
you can use map here. 
let parsed_value_list = parsed_value.map(item => parseValue(item));

This is more functional programming and we should use this pattern more.

Reply:
Done